src/OrleansZooKeeperUtils/MembershipSerializerSettings.cs

```diff
@@ -46,7 +46,7 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                 {
                     SiloAddress = jo["SiloAddress"].ToObject<SiloAddress>(serializer),
                     HostName = jo["HostName"].ToObject<string>(),
-                    InstanceName = jo["InstanceName"].ToObject<string>(),
+                    SiloName = jo["SiloName"].ToObject<string>(),
```

Comment:
This will throw when "SiloName" doesn't exist. I'll need to improve serialization here to account for this in an elegant way. In the meantime, it can just be 
```csharp
SiloName = (jo["SiloName"]??jo["InstanceName"]).ToObject<string>(),
```